pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/module/RelatedModuleServlet.java

```diff
@@ -0,0 +1,95 @@
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static com.redhat.pantheon.conf.GlobalConfig.DEFAULT_MODULE_LOCALE;
+import static com.redhat.pantheon.servlet.ServletUtils.paramValue;
+import static com.redhat.pantheon.servlet.ServletUtils.paramValueAsLocale;
+import static com.redhat.pantheon.servlet.ServletUtils.writeAsJson;
+import static javax.servlet.http.HttpServletResponse.SC_NOT_FOUND;
+
+/**
+ * Get operation to render a Released Module's related module list in JSON format.
+ * Only two parameters are expected in the Get request:
+ * 1. locale - Optional; indicates the locale that the module content is in, defaulted to en-US
+ * 2. module_id - indicates the uuid string which uniquely identifies a module
+ *
+ * The url to GET a request from the server is /api/module/related
+ * Example: <server_url>/api/module/related?locale=en-us&module_id=xyz
+ * The said url is accessible outside of the system without any authentication.
+ *
+ * @author Ben Radey
+ */
+@Component(
+        service = Servlet.class,
+        property = {
+                Constants.SERVICE_DESCRIPTION + "=Servlet to facilitate GET operation which accepts locale and module uuid to output module relationships",
+                Constants.SERVICE_VENDOR + "=Red Hat Content Tooling team"
+        })
+@SlingServletPaths(value = "/api/module/related")
+public class RelatedModuleServlet extends SlingSafeMethodsServlet {
+
+    private final Logger log = LoggerFactory.getLogger(RelatedModuleServlet.class);
+
+    @Override
+    protected void doGet(@NotNull SlingHttpServletRequest request, @NotNull SlingHttpServletResponse response) throws ServletException, IOException {
+        Locale locale = paramValueAsLocale(request, "locale", DEFAULT_MODULE_LOCALE);
+        String uuidParam = paramValue(request, "module_id", "");
+
+        StringBuilder query = new StringBuilder("select * from [pant:module] as module WHERE module.[jcr:uuid] = '")
+                .append(uuidParam)
+                .append("'");
+
+        JcrQueryHelper queryHelper = new JcrQueryHelper(request.getResourceResolver());
+        try {
+            Stream<Resource> resultStream = queryHelper.query(query.toString());
+
+            Optional<Resource> firstResource = resultStream.findFirst();
+            if(!firstResource.isPresent()) {
+                response.sendError(SC_NOT_FOUND, "Requested content not found.");
+            }
+
+            resultStream = queryHelper.query("select * from [pant:module] as module", 3, 0);
+
+            List<Map> related = new ArrayList<>();
+            resultStream.map(r -> r.adaptTo(Module.class))
+                    .forEach(module -> {
+                            Map<String, String> m = new HashMap<>();
+                            m.put("title", module.getModuleLocale(locale).getVersion("1").metadata().get().title().get());
+                            m.put("url", "https://www.redhat.com");
```

Comment:
I am wondering why do we set the url value to https://www.redhat.com?

Reply:
I wanted it to be something that was easy to recognize as wrong. If we made it something realistic, someone might accidentally think that the related modules list was legitimate.